File: frontend/src/types/WorkoutSession.ts
import { User } from '@/types/User';
import { Workout } from '@/types/Workout';

export interface WorkoutSession {
  id: number;
  workoutId: number;
  workout?: Workout;
  traineeId: number;
  trainee?: User;
  startTime: string;
  endTime: string;
  perceivedDifficulty?: number;
  notes?: string;
}

export interface CreateWorkoutSessionRequest {
  workoutId: number;
  traineeId: number;
  startTime: string;
  endTime: string;
  perceivedDifficulty?: number;
  notes?: string;
}

export interface UpdateWorkoutSessionRequest {
  perceivedDifficulty?: number;
  notes?: string;
}

export interface WorkoutSessionQueryParams {
  pageNumber?: number;
  pageSize?: number;
  searchTerm?: string;
  workoutId?: number;
  startDateFrom?: string;
  startDateTo?: string;
  endDateFrom?: string;
  endDateTo?: string;
  minPerceivedDifficulty?: number;
  maxPerceivedDifficulty?: number;
  sortBy?: string;
  sortOrder?: 'asc' | 'desc';
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
  hasPrevious: boolean;
  hasNext: boolean;
}

File: backend/Easygym.Domain/Entities/WorkoutSession.cs
using System.ComponentModel.DataAnnotations;

namespace Easygym.Domain.Entities
{
    public class WorkoutSession
    {
        public int Id { get; set; }

        [Range(1, int.MaxValue)]
        public required int WorkoutId { get; set; }
        public Workout? Workout { get; set; }

        [Range(1, int.MaxValue)]
        public required int TraineeId { get; set; }
        public User? Trainee { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        [Range(1, 10)]
        public int? PerceivedDifficulty { get; set; }

        [StringLength(1000)]
        public string? Notes { get; set; }
    }
}
